File: app/src/main-java/java/com/example/MainActivity.java
package com.example;
import android.app.Activity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import com.example.databinding.ActivityMainBinding;
import io.mdp43140.ael.ErrorLogger;
public class MainActivity extends Activity {
	@Override protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
		binding.btn1.setOnClickListener(v -> {
			try {
				// do something
				throw new Exception("Exception caught with try-catch goes through ErrorLogger...");
			} catch (Exception e) {
				ErrorLogger.instance.handleError(e);
			}
		});
		binding.btn2.setOnClickListener(v -> {
			// do something
			System.out.print(1 / 0);
		});
		setContentView(binding.getRoot());
	}
}

File: app/src/main/java/com/example/MainActivity.kt
package com.example
import android.app.Activity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import com.example.databinding.ActivityMainBinding
import io.mdp43140.ael.ErrorLogger
class MainActivity: Activity(){
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		ActivityMainBinding.inflate(layoutInflater).apply {
			btn1.setOnClickListener {
				try {
					// do something
					throw Exception("Exception caught with try-catch goes through ErrorLogger...")
				} catch (e: Exception){
					ErrorLogger.instance?.handleError(e)
				}
			};
			btn2.setOnClickListener {
				// do something
				throw Exception("Uncaught Exception goes through ErrorLogger...")
			};
			setContentView(root)
		}
	}
}
